fw/ss_skb.c

```diff
@@ -1571,3 +1571,28 @@ ss_skb_add_frag(struct sk_buff *skb_head, struct sk_buff *skb, char* addr,
 
 	return 0;
 }
+
+/* Using @split_point transform the remaining linear portion of original @skb
+ * to the firsst fragment of the same SKB. Existing fragments of @skb
+ * would moved to next SKB if necessary inside __split_linear_data().
+ */
+int
+ss_skb_linear_transform(struct sk_buff *skb_head, struct sk_buff *skb,
+			unsigned char *split_point)
+{
+	int fpos, r;
+	TfwStr _;
+
+	if (!split_point) {
+		/* Usage of linear portion of SKB is not expected */
+		skb_pull(skb, skb_headlen(skb));
```

Comment:
IIUC skb that has "payload" for instance http headers will contain these headers in "tail part". Such skbs looks like this: `MAX_HEADERS + HTTP_HEADERS`, where `MAX_HEADERS` was reserved using `skb_reserve()` and `HTTP_HEADERS` placed into skb using `skb_put`. It implies the part contains `HTTP_HEADERS` must be removed using `ss_skb_put`, like you doing in code below. Also would be better to set `tail_lock` flag.